libsolidity/lsp/VFS.h

```diff
@@ -0,0 +1,101 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <libsolidity/lsp/LSPTypes.h>
+#include <libsolidity/lsp/TextBuffer.h>
+
+#include <deque>
+#include <map>
+#include <string>
+#include <ostream>
+#include <vector>
+
+namespace solidity::lsp::vfs
+{
+
+using TextLines = std::deque<std::string>;
+
+class File
+{
+public:
+	File(std::string path, std::string _languageId, int _version, std::string _text);
+
+	// readers
+	std::string const& path() const noexcept { return m_path; }
+	std::string const& languageId() const noexcept { return m_languageId; }
+	constexpr int version() const noexcept { return m_version; }
+	std::string const& contentString() const { return m_buffer.data(); }
+	TextBuffer const& buffer() const noexcept { return m_buffer; }
+
+	// modifiers
+	constexpr void setVersion(int _version) noexcept { m_version = _version; }
+	void erase(LineColumnRange _range);
+	void modify(LineColumnRange _range, std::string const& _replacementText);
+	void replace(std::string const& _replacementText);
+
+	static TextLines splitLines(std::string const& _text);
```

Comment:
Why a deque? In which way is this different from boost::string::split?

Reply:
The LSP's design specification should not include source files of more than 1k lines. I really think all those operations can efficiently performed an simple strings on the fly.

Reply:
I keep forgetting about boost API. I've deleted this.